gina-alaska/viirs-snow-metrics
# Add snow metric computation module

This PR accomplishes a few things:
 - Adds a `compute_snow_metrics.py` module to compute three initial snow metrics.
 - Generalizes "watermasking" to "compute_masks"
 - Updates the README with information about the snow metrics that are computed so far.
 - Adds a draft "User Guide" to track output specs and semantics

To test this PR:
- Read the README - it has been updated! Set your environment variables accordingly.
- Active your `viirs-snow` conda env.
 - If you already have a batch of downloaded and pre-processed data, and you've computed the masks you can skip the following steps. Otherwise do this stuff again:
   - Go ahead and wipe your pre-existing download data and preprocessed datasets. 
   - Run `python download.py` and then go work on something else or take a walk, this might take all day. Give the log file a quick scan. Examine the download: `ls -lhrt $INPUT_DIR/$SNOW_YEAR` or similar.
   - Run `python preprocess.py $tile_id` and examine ethe output directory and scan the log file. Maybe try `h10v02` or `h12v02`. I've been testing `h11v02`.
   - Run `python compute_masks.py $tile_id` and examine the output directory and scan the log file.
- With data prepped and mask files computed, you should now you should be ready to compute some actual snow metrics for a single tile!
  - Run `compute_snow_metrics.py $tile_id`
  - Verify that three GeoTIFFs, one per metric, were created in the output location you expect.
  - Examine the metric GeoTIFFs in a GIS. Bring them to your local machine with something like `scp -r "$USER"@elephant.gina.alaska.edu://export/datadir/$USER"_viirs_snow/$SCRATCH_DIR/$SNOW_YEAR/single_metric_geotiffs/ .` Examine some individual grid cell values. Look at some spatial patters. Do the sdata seem to pass the eyeball test?
  - Try `gdalinfo` on these GeoTIFFs. Is there a metadata tag that make these data self-describing beyond just the name of the file?
